app/Models/Post.php

```diff
@@ -49,11 +49,17 @@ public function signup()
     }
 
     /**
-     * Each post has many reactions.
+     * Query for total reactions for a post.
+     *
+     * @param int $postableId
+     * @param int $postableType
+     * @return int total count
      */
-    public function reactions()
+    public static function withReactionCount($postableId, $postableType)
     {
-        return $this->hasMany(Reaction::class);
+        return self::withCount(['reactions' => function ($query) use ($postableId, $postableType) {
```

Comment:
also for the future, here is some code I was fidgeting with:
```
    /**
     * Query for total reactions for a photo.
     *
     * @param int $reactionableId
     * @return int total count
     */
    public static function withReactionCount($reactionableId)
    {

        // $posts = Post::withCount(['votes', 'comments' => function ($query) {
        //     $query->where('content', 'like', 'foo%');
        // }])->get();

        // return self::withCount('reactions')->get();

        // this returns all Photos in an array that we can iterate through and call `->reaction_count` on each
        return self::withCount(['reactions', 'id' => function ($query) use ($reactionableId) {
            $query->where('id', $reactionableId);
        }])->get();

        // this gets us the correct photo but we can't call ->reactions_count on it
        // return self::where('id', $reactionableId)->withCount('reactions')->get();

    }
```
The problem that I was running into was returning the specific photo queried by the `reactionableId` - I could only get this to return all Photos and then call `->reaction_count` on one photo at a time (e.g. `$photos[0]->reaction_count`) so we'd still have to query again by `reactionableId`.